BepInEx/KeyboardShortcut.cs

```diff
@@ -0,0 +1,162 @@
+        /// <param name="control">Should Control be held down?</param>
+        /// <param name="alt">Should Alt be held down?</param>
+        /// <param name="shift">Should Shift be held down?</param>
+        public KeyboardShortcut(KeyCode key, bool control = false, bool alt = false, bool shift = false)
+        {
+            Key = key;
+            Control = control;
+            Alt = alt;
+            Shift = shift;
+        }
+
+        public KeyboardShortcut()
+        {
+        }
+
+        public event PropertyChangedEventHandler PropertyChanged;
+
+        public bool Alt
+        {
+            get
+            {
+                return alt;
+            }
+
+            set
+            {
+                alt = value;
+                PropertyChanged?.Invoke(this, new PropertyChangedEventArgs(nameof(Alt)));
+            }
+        }
+
+        public bool Control
+        {
+            get
+            {
+                return control;
+            }
+
+            set
+            {
+                control = value;
+                PropertyChanged?.Invoke(this, new PropertyChangedEventArgs(nameof(Control)));
+            }
+        }
+
+        public KeyCode Key
+        {
+            get
+            {
+                return key;
+            }
+
+            set
+            {
+                key = value;
+                PropertyChanged?.Invoke(this, new PropertyChangedEventArgs(nameof(Key)));
+            }
+        }
+
+        public bool Shift
+        {
+            get
+            {
+                return shift;
+            }
+
+            set
+            {
+                shift = value;
+                PropertyChanged?.Invoke(this, new PropertyChangedEventArgs(nameof(Shift)));
+            }
+        }
+
+        public static KeyboardShortcut Deserialize(string str)
+        {
+            try
+            {
+                var parts = str.Split(' ').Select(x => int.Parse(x)).ToArray();
+                return new KeyboardShortcut((KeyCode)parts[0], parts[1] == 1, parts[2] == 1, parts[3] == 1);
+            }
+            catch (SystemException ex)
+            {
+                BepInLogger.Log("Failed to read keybind from settings: " + ex.Message);
+                return null;
+            }
+        }
+
+        public string Serialize()
+        {
+            return $"{(int)Key} {(Control ? 1 : 0)} {(Alt ? 1 : 0)} {(Shift ? 1 : 0)}";
+        }
+
+        /// <summary>
+        /// Check if the main key was just pressed (Input.GetKeyDown), and specified modifier keys are all pressed
+        /// </summary>
+        public bool IsDown()
+        {
+            return Input.GetKeyDown(Key) && ModifierKeyTest();
+        }
+
+        /// <summary>
+        /// Check if the main key is currently held down (Input.GetKeyDown), and specified modifier keys are all pressed
+        /// </summary>
+        public bool IsPressed()
+        {
+            return Input.GetKeyDown(Key) && ModifierKeyTest();
```

Comment:
What's the difference from `IsDown`? Just the name? Or should this be instead the following?

```csharp
return Input.GetKey(Key) && ModifierKeyTest();
```

More info: [Input.GetKey](https://docs.unity3d.com/ScriptReference/Input.GetKey.html).

Reply:
You are correct, it was supposed to be Input.GetKey